plugins/omar-wfs-plugin/grails-app/services/omar/wfs/WebFeatureService.groovy

```diff
@@ -664,11 +693,26 @@ println '*'*40
               }
 
               // the footprint geometry
-              mkp.yieldUnescaped( feature.ground_geom.getKml() )
+              if  ( feature.ground_geom )
+              {
+                mkp.yieldUnescaped( feature.ground_geom.getKml() )
+              }
+              else
+              {
+                MultiGeometry {
+                  Polygon {
+                    outerBoundaryIs {
+                      LinearRing {
+                        coordinates(feature.geometry.coordinates[0][0].collect { it.join(',') }.join(' '))
+                      }
+                    }
+                  }
+                }
+              }
 
               Snippet()
 
-              switch (feature.sensor_id) {
+              switch (feature.sensor_id ?: feature.properties.sensor_id) {
```

Comment:
Maybe naive, but I don't see the end bracket for this switch in the git diff

Reply:
The end bracket doesn't show up in the diff because the end bracket didn't change. It's there, just not in the diff.